test/unit/routes/static.test.ts

```diff
@@ -0,0 +1,47 @@
+import { promises as fs } from "fs"
+import * as path from "path"
+import { tmpdir } from "../../utils/helpers"
+import * as httpserver from "../../utils/httpserver"
+import * as integration from "../../utils/integration"
+
+describe("static", () => {
+  let codeServer: httpserver.HttpServer | undefined
+  let testFile: string | undefined
+  let testFileContent: string | undefined
+
+  beforeAll(async () => {
+    const testDir = await tmpdir("static")
+    testFile = path.join(testDir, "test")
+    testFileContent = "static file contents"
+    await fs.writeFile(testFile, testFileContent)
+  })
+
+  afterEach(async () => {
+    if (codeServer) {
+      await codeServer.close()
+      codeServer = undefined
+    }
+  })
+
+  it("/static unauthed", async () => {
+    codeServer = await integration.setup(["--auth=none"], "")
+    let resp = await codeServer.fetch("/static")
+    expect(resp.status).toBe(404)
+
+    resp = await codeServer.fetch(`/static/-${testFile}`)
+    expect(resp.status).toBe(200)
+
+    const content = await resp.text()
+    expect(content).toStrictEqual(testFileContent)
+  })
+
+  it("/static authed", async () => {
```

Comment:
nit: testname should explain what the test is testing for -> 
```suggestion
  it("should return a 401 if authentication is enabled and user requests files", async () => {
```